packages/socks-proxy-agent/src/index.ts

```diff
@@ -142,7 +142,7 @@ export class SocksProxyAgent extends Agent {
 			timeout: timeout ?? undefined,
 		};
 
-		const cleanup = (tlsSocket?: tls.TLSSocket) => {
+		const cleanup = (tlsSocket?: net.Socket) => {
```

Comment:
I think this change isn't necessary, if we remove the return annotation from `upgradeSocketToTls()` like suggested above.